File: AppGenerateFiles/helpdesk/Model/CaseTable_Tareas.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class CaseTable_Tareas : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id_Tarea { get; set; }
       public string? Titulo { get; set; }
       public int? Id_TareaPadre { get; set; }
       public int? Id_Case { get; set; }
       public string? Descripcion { get; set; }
       public string? Estado { get; set; }
       public DateTime? Fecha_Inicio { get; set; }
       public DateTime? Fecha_Finalizacion { get; set; }
       public DateTime? Fecha_Inicio_Proceso { get; set; }
       public DateTime? Fecha_Finalizacion_Proceso { get; set; }
       [ManyToOne(TableName = "CaseTable_Case", KeyColumn = "Id_Case", ForeignKeyColumn = "Id_Case")]
       public CaseTable_Case? CaseTable_Case { get; set; }
       [ManyToOne(TableName = "CaseTable_Tareas", KeyColumn = "Id_Tarea", ForeignKeyColumn = "Id_TareaPadre")]
       public CaseTable_Tareas? CaseTable_Tareas { get; set; }
       [OneToMany(TableName = "CaseTable_Calendario", KeyColumn = "Id_Tarea", ForeignKeyColumn = "Id_Tarea")]
       public List<CaseTable_Calendario>? CaseTable_Calendario { get; set; }
       [OneToMany(TableName = "CaseTable_Evidencias", KeyColumn = "Id_Tarea", ForeignKeyColumn = "Id_Tarea")]
       public List<CaseTable_Evidencias>? CaseTable_Evidencias { get; set; }
       [OneToMany(TableName = "CaseTable_Participantes", KeyColumn = "Id_Tarea", ForeignKeyColumn = "Id_Tarea")]
       public List<CaseTable_Participantes>? CaseTable_Participantes { get; set; }
       [OneToMany(TableName = "CaseTable_Tareas", KeyColumn = "Id_Tarea", ForeignKeyColumn = "Id_TareaPadre")]
       public List<CaseTable_Tareas>? CaseTable_Tareas { get; set; }
   }
}

File: AppGenerateFiles/helpdesk/FrontModel/CaseTable_Tareas.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { WAjaxTools } from "../../WDevCore/WModules/WComponentsTools.js";
//@ts-ignore
import { ModelProperty } from "../../WDevCore/WModules/CommonModel.js";
import { CaseTable_Case }  from './CaseTable_Case.js'
import { CaseTable_Tareas }  from './CaseTable_Tareas.js'
import { CaseTable_Calendario }  from './CaseTable_Calendario.js'
import { CaseTable_Evidencias }  from './CaseTable_Evidencias.js'
import { CaseTable_Participantes }  from './CaseTable_Participantes.js'
import { CaseTable_Tareas }  from './CaseTable_Tareas.js'
class CaseTable_Tareas extends EntityClass {
   constructor(props) {
       super(props, 'EntityHelpdesk');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id_Tarea;
   /**@type {String}*/ Titulo;
   /**@type {String}*/ Descripcion;
   /**@type {String}*/ Estado;
   /**@type {Date}*/ Fecha_Inicio;
   /**@type {Date}*/ Fecha_Finalizacion;
   /**@type {Date}*/ Fecha_Inicio_Proceso;
   /**@type {Date}*/ Fecha_Finalizacion_Proceso;
   /**@type {CaseTable_Case} ManyToOne*/ CaseTable_Case;
   /**@type {CaseTable_Tareas} ManyToOne*/ CaseTable_Tareas;
   /**@type {Array<CaseTable_Calendario>} OneToMany*/ CaseTable_Calendario;
   /**@type {Array<CaseTable_Evidencias>} OneToMany*/ CaseTable_Evidencias;
   /**@type {Array<CaseTable_Participantes>} OneToMany*/ CaseTable_Participantes;
   /**@type {Array<CaseTable_Tareas>} OneToMany*/ CaseTable_Tareas;
}
export { CaseTable_Tareas }
